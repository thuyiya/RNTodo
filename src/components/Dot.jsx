import React from 'react';
import { View } from 'react-native';
import { theme } from '../constants';

const Dot = () => {
    return <View style={{
        backgroundColor: theme.colors.primary,
        height: 20,
        width: 20,
        borderRadius: 10
    }}/>
}

export default Dot