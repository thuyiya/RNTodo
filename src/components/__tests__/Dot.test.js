import React from "react";
import renderer from "react-test-renderer";
import Dot from "../Dot";

describe("<Dot />", () => {
  it("has no children", () => {
    const tree = renderer.create(<Dot />).toJSON();
    expect(tree.children).toBe(null);
  });

  it("has 20px circle area", () => {
    const tree = renderer.create(<Dot />).toJSON();
    expect(tree.props.style.height).toBe(20);
    expect(tree.props.style.width).toBe(20);
    expect(tree.props.style.borderRadius).toBe(10);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Dot />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
