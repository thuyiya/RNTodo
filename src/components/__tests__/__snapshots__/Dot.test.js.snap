// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dot /> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#0145a6",
      "borderRadius": 10,
      "height": 20,
      "width": 20,
    }
  }
/>
`;
